kernel/src/arch/x86_64/pci.rs

```diff
@@ -0,0 +1,214 @@
+use super::port;
+use crate::driver;
+use alloc::vec::Vec;
+
+const CONFIG_ADDRESS: u16 = 0xCF8;
+const CONFIG_DATA: u16 = 0xCFC;
+const ENABLE: u32 = 1 << 31;
+
+const VENDOR_ID: u8 = 0;
+pub const COMMAND: u8 = 4;
+const CLASS: u8 = 10;
+const BAR: u8 = 16;
+
+#[derive(Debug, Clone, Copy)]
+pub struct Device {
+    pub bus: u8,
+    pub slot: u8,
+}
+
+impl Device {
+    /// Creates a PCI device at bus and slot address
+    /// Returns None if device does not exist
+    pub fn new(bus: u8, slot: u8) -> Option<Device> {
```

Comment:
```suggestion
    pub fn try_new(bus: u8, slot: u8) -> Option<Device> {
```
nit: Since the function is "faillible"

Reply:
Great idea thanks !